lib/src/matchers.rs

```diff
@@ -506,6 +513,132 @@ impl<V: Debug> Debug for RepoPathTree<V> {
     }
 }
 
+#[derive(Clone, Debug)]
+struct IgnoreWrapper(ignore::gitignore::Gitignore);
+
+impl Default for IgnoreWrapper {
+    fn default() -> Self {
+        Self(ignore::gitignore::Gitignore::empty())
+    }
+}
+
+/// Matches file paths with glob patterns.
+///
+/// Patterns are provided as `(dir, pattern)` pairs, where `dir` should be the
+/// longest directory path that contains no glob meta characters, and `pattern`
+/// will be evaluated relative to `dir`.
+#[derive(Clone, Debug)]
+pub struct GitAttributesMatcher {
+    tree: RepoPathTree<IgnoreWrapper>,
+}
+
+#[derive(Debug, Error)]
+pub enum GitAttributesMatcherError {
+    #[error("failed to parse .gitattributes file")]
+    GitIgnoreError(#[from] ignore::Error),
+    #[error("failed to read .gitattributes file")]
+    ReadError(#[from] io::Error),
+    #[error("invalid pattern in .gitattributes: {0}")]
+    InvalidPattern(String),
+}
+
+fn parse_gitattributes(
```

Comment:
There is a [gix-attributes](https://crates.io/crates/gix-attributes) crate, probably it's better to use it?

Reply:
We already use `gix`, so enabling some additional features should be fine.